test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -265,6 +265,59 @@ void testExprSplitWithMask01() {
   ExpectAllNear(c_v, c_ref, 1e-5);
 }
 
+void testSplitWithTailWithLoopOptions() {
+  KernelScope kernel_scope;
+  const int M = 21;
+  Buffer a_buf("a", kFloat, {M});
+  Buffer b_buf("b", kFloat, {M});
+  Tensor* tensor = Compute("f", {{M, "m"}}, [&](const ExprHandle& m) {
+    return a_buf(m) + b_buf(m) + 1.0f;
+  });
+  For *outer, *inner, *tail;
+
+  LoopNest l({tensor});
+  auto loops = NodeFinder<For>::find(l.root_stmt());
+  ASSERT_GT(loops.size(), 0);
+  l.setGPUBlockIndex(loops[0], LoopOptions::IDX_Y);
+  l.splitWithTail(loops[0], 4, &outer, &inner, &tail);
+  ASSERT_NE(outer, nullptr);
+  ASSERT_NE(inner, nullptr);
+  ASSERT_NE(tail, nullptr);
+
+  // Outer loop carries loop axis bindings.
+  ASSERT_TRUE(outer->loop_options().is_gpu_block_index());
+  ASSERT_EQ(outer->loop_options().gpu_block_index(), LoopOptions::IDX_Y);
+
+  // Inner loop has none.
+  ASSERT_TRUE(inner->loop_options().isDefault());
+
+  // Tail loop has none.
```

Comment:
I wonder if it's always correct to drop the properties in the tail loop: an alternative could be to error out if the original loop has any loop options.

Reply:
Yeah it's an interesting question, I'm not 100% sure. In practice we generally SplitWithMask on GPU to avoid this consideration. I think personally losing the axis bindings in the tail makes the most sense to me, and it improves composability so thats what I went with. Open to be convinced though.

Reply:
I prefer explicitly erroring out in this case, similarly to what we decided to do in inlining. Overall, I think a good guideline could be "whenever several reasonable outcomes are possible from a transformation, prefer to error out rather than silently picking one of the options". I don't think it would reduce composability, as one could still achieve the desired results with a properly ordered transforms, but it could reduce a potential surprise factor. What are your thoughts on this?

Reply:
Sorry @ZolotukhinM I missed this comment. That is another way to look at it, for sure. Similar to inlining it would in practice create an ordering of which transforms are done at which time - ie. do loop transforms first, then do axis bindings. I think if we are starting to go down that road we should make it explicit - but the thought does make me a bit uncomfortable.